### ISSUE TITLE
[go_router] Resume navigation after redirect (e.g. post-login)

### ISSUE THREAD

**Issue by wiradikusuma** [flutter/flutter#120371]
I might be missing something obvious, but I can't find any example of resuming navigation after a redirect.

I have a route, `/authenticated/page`, that can only be viewed once the user is logged in.

go_router's [official redirection example][1] shows how to do such "login gate", but it redirects to `/`. 

How do I resume navigation after redirect (e.g. login)? I.e.:

`/` → click link to `/authenticated/page` → blocked by login page → successful login → opens `/authenticated/page`

(On the web, typically we pass param, e.g. `/login?next=/profile`—how do I do that for redirects?)

  [1]: https://github.com/flutter/packages/blob/main/packages/go_router/example/lib/redirection.dart

**Comment by huycozy** [comment]
Hi @wiradikusuma 
It seems your question is best suited for support channels like StackOverflow, so please ask the same there for solutions if any. If you are proposing a change, please use the appropriate new issue template and fill out the requested details so that we can address this issue properly.
Closing from here as this is more of a `how-to` question rather than a bug or feature request.

**Comment by wiradikusuma** [comment]
Hi @huycozy 

I actually did ask this in [Stack Overflow](https://stackoverflow.com/questions/75377395/flutter-go-router-resume-navigation-after-redirect-e-g-login), but point taken.

However, can the example please be updated with how to pass param in the redirect?

**Comment by huycozy** [comment]
@wiradikusuma It looks like you can achieve this by using `Provider`, like the sample code from OP at this proposal: https://github.com/flutter/flutter/issues/112915. Please try it to see if it works for you.
